react/components/BuyButton/index.js

```diff
@@ -45,44 +42,60 @@ export class BuyButton extends Component {
   }
 
   handleAddToCart = async () => {
-    const { skuItems, isOneClickBuy, orderFormContext, push } = this.props
+    const { addToCart, skuItems, isOneClickBuy } = this.props
     this.setState({ isAddingToCart: true })
 
-    const variables = {
-      items: skuItems.map(skuItem => {
-        const { skuId } = skuItem
+    if (isOneClickBuy) location.assign(CONSTANTS.CHECKOUT_URL)
+
+    let toastMessage = null
+    try {
+      const minicartItems = skuItems.map(skuItem => {
+        const {
+          skuId: id,
+          variant: skuName,
+          price: sellingPrice,
+          detailUrl,
+          imageUrl,
+          quantity,
+          seller,
+          name,
+          options,
+          listPrice,
+        } = skuItem
+
         return {
-          id: parseInt(skuId),
+          detailUrl,
+          id,
+          imageUrl,
+          listPrice,
+          name,
+          quantity,
+          sellingPrice,
+          skuName,
```

Comment:
Why did you lose the `pick`? It got more verbose

Reply:
ramda over JS... Someday I'll understand that

Reply:
It's not about that, any helper library would do in that case, really.

You turned a transformation of a data structure to return into 1) destructuring the initial DS 2) creating another DS with those destructured props. 

Despite way less LOC's, it's more error prone because you are duplicating the name writing of a bunch of names. Without typechecking or unit tests, it becomes easy to future updates forget something and, if not, it is clearly takes more time

Reply:
Are there more properties being passed than those one listes in `propTypes`? If so, [omit](https://ramdajs.com/docs/#omit) would be even less verbose.

Reply:
I partially agree, because about this type check, we'll do it in this project. And it won't cover the `pick` function because they are strings. So, if anyone changes some string value, the type check will do nothing. Anyway, I've used the `pick` function, as you asked, senpai :)

Reply:
About, omit... Maybe there are more properties. But even if there aren't, I want to make sure only these selected properties go to the minicart. I don't wanna risk someone adding stuff and breaking the link-state.

Reply:
Ok. 

About the typechecking of `pick`, that is not actually true, as you can see [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ramda/index.d.ts#L1751) you can (and our platforms will enforce you of that) pass the type parameters expected for `pick`

Reply:
Oh cool! I just feel a little bit curious about this `If the key does not exist, the property is ignored`. Because if I mistyped a property, It will warn me or just ignore it? I'll play with it someday ;)